src/Mod/Draft/draftguitools/gui_edit_draft_objects.py

```diff
@@ -123,7 +132,43 @@ def update_preview_object(self, edit_command, obj, node_idx, v):
             if obj.Closed:
                 pointList.append(pointList[0])
             edit_command.ghost.updateFromPointlist(pointList)
-        
+
+    def add_point(self, edit_command, obj, pos):
+        """Add point to obj.
+        """
+        # self.setSelectState(obj, True)
+        selobjs = Gui.ActiveDocument.ActiveView.getObjectsInfo((pos[0],pos[1]))
```

Comment:
Should this be done here? It might make more sense to just use `obj` here? I realize that this is just the original code, but given that `obj` is now passed, it makes this "double work" stand out more.

I do think there will be a change in behaviour if you use `obj`, though. From the looks of it, the old intended behaviour is to add the point to *all* edited objects at the current location, not just the one `obj`. However, it seems that that behavior is currently bugged (if you add have intersecting lines, edit one of them, add a point at the intersection, you'll get two points in the edited line, not one. Also, if you edit a wire that has two segments very close together and add a point between them, it seems the result is even more messed up). Also, the current behavior does not really make sense to me: The "add point" command has been added to the menu based on `obj`, not *all* objects under the cursor, so it would make sense to only modify `obj` here (and in fact, now that add point handling is split among different draft object types, it would even be hard to add a point to e.g. both a wire and a spline in the same action).

Looking more closely at your code, you actually already change this behavior by checking `if obj.Name != info["Object"]:`. This means that you could really just leave out the for loop and just act on `obj`, which would result in the same behavior as your current code, AFAICS.

Leaving out the call to `getObjectsInfo` and the loop, also removes most duplicate code between the `add_points` methods that you introduced (that I was going to make a comment about, but that no longer seems necessary).

Finally, I would suggest you make these changes to the original `addPoint` method in a separate commit, by passing the `obj` to that `addPoint` and dropping the loop there already. This cleanly separates the behavior change in a small and easy to review commit, and the bigger generalization commit can then just move around code without changing any behavior.

Reply:
Reading the code a bit more closely, I think that the call to `getObjectsInfo` might actually still be needed, since i *think* it does not only figure out which object to act on, but also figures out which part of the object is nearest and what the exact coordinates are (i.e. `pos` might be right next to the line, while `info[x/y/z]` is always *on* the line.

Given that, I think that it would still be good to reduce the duplicate code by creating something like `getSpecificObjectInfo(obj, (x,y))` that essentially does what your code does now: Call `getObjectsInfo` and pick out only the results that point to `obj` (maybe even just *one* result which is closest, I think you rarely want to add multiple points when line segments are close together). Maybe it could also already create a Vector out of `info[x/y/z]` since that seems to be used everywhere as well. Adding this new function (and the associated behavior change) could still be done first in `addPoint` in the separate commit I mentioned.

Reply:
Yes, you got it perfectly. I was thinking that we can add another argument in get_edit_obj_context_menu(self, edit_command, obj, position) and pass the function the real vector of the point in the 3d scene, so this stuff can just be ignored by the object code and handled completely in the Draft_Edit command. I'll give it a try!

Reply:
I'm not sure I understand what other argument you mean? Isn't the `pos` already the real vector? Or is that the 2D vector and you mean to do the `getObjectsInfo` in `display_tracker_menu` and pass the 3D vector result, i.e. from `info[x/y/z]`? If the latter, then I think you'd also need to pass `info`, since IIRC some functions use more from info than just the vector.

Reply:
I meant to pass the 3D vector result, but it's true we would also need to pass `info`, so better to leave the `pos[x/y]` at least for the moment.

Reply:
> Given that, I think that it would still be good to reduce the duplicate code by creating something like getSpecificObjectInfo(obj, (x,y)) that essentially does what your code does now: Call getObjectsInfo and pick out only the results that point to obj

Reading this again perhaps you are right; 
for reference this is a result of `getObjectInfo`:   `{'x': 123.58872985839844, 'y': -6.468395233154297, 'z': 0.0, 'Document': 'Unnamed', 'Object': 'Wire', 'Component': 'Edge3'}`

the `getSpecificObjectInfo(obj, (x,y))` method could return:
- or the entire obj info;
- or the clicked subobject and the 3D vector result (I guess this would cover all the usecases I can imagine, because we already know the object and the callback mechanism is bound to the view, so I guess no chance to be activated on another document).